core/iwasm/libraries/libc-wali/wali.c

```diff
@@ -407,7 +407,11 @@ long
 wali_syscall_open(wasm_exec_env_t exec_env, long a1, long a2, long a3)
 {
     long args[6] = { a1, a2, a3 }; // strace
-
+    //security check
+    if (strncmp((char*)MADDR(a1), "/proc/self/mem", 15) == 0) {
+        printf("Unpermitted attempt to open /proc/self/mem.");
+        RETURN(-1);
+    }
```

Comment:
* Again, can you make these use the logging only on the verbose setting? It also seems like you could abstract this into a function that is invoked at every callsite